### PR TITLE
Refactor/scantask get facts

### PR DESCRIPTION
**Pull request by bruno-fs**
Bunch of refactoring/optimizations related to `ScanTask` model.

### CODE REVIEW

**Review by gtanzillo — APPROVED** [review]
LGTM 👍

File: quipucords/api/scantask/test_scantask_get_facts.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    return systems
+
+
+@pytest.fixture
+def inspection_scantask(raw_facts_list):
+    """Scantask instance mapped to raw_facts_list."""
+    task_inspection_result = TaskInspectionResult.objects.create(
+        job_inspection_result=JobInspectionResult.objects.create()
+    )
+    for i, facts in enumerate(raw_facts_list):
```
Comment (nicolearagao) [inline quipucords/api/scantask/test_scantask_get_facts.py:42]: lint is complaining about the variable i, saying it doesn comply with the snake case rule. Maybe changing it to index would mute this warning?

Comment (bruno-fs) [reply quipucords/api/scantask/test_scantask_get_facts.py:42]: Mmm maybe this is a side effect from adding "db" to the list of "good-words"

Comment (bruno-fs) [reply quipucords/api/scantask/test_scantask_get_facts.py:42]: yeah.. here it is: https://github.com/PyCQA/pylint/blob/main/pylintrc#L234

Comment (bruno-fs) [reply quipucords/api/scantask/test_scantask_get_facts.py:42]: funny it isn't complaining about the `_` :eyes:

Comment (bruno-fs) [reply quipucords/api/scantask/test_scantask_get_facts.py:42]: Insted of changing the code I included the "good names" from the original pylintrc. what do u think, @nicolearagao ?

Comment (nicolearagao) [reply quipucords/api/scantask/test_scantask_get_facts.py:42]: good idea

**Review by nicolearagao — APPROVED** [review]
LGTM

**Pull request merged**